### DISCUSSION TITLE
How can we send videos and GIFs that can be watched in the chat and have a preview

### DISCUSSION THREAD

**Discussion by okaeiz** [kianahmadian/python-bale-bot#18] [Q&A] [answered]
When I send a video or a GIF, no matter it has a caption or not, the video will be sent to the users as a document. Thus, they cannot see its preview.
If the file is a video, it may be acceptable but when it is a GIF, it should play over and over again in the chat; without having to download it like a video file (.mp4).
Is there any solution?

**Comment by kian-ahmadian** [comment]
I do not know about this. I will check and announce here.

**Comment by kian-ahmadian** [comment, marked as answer]
@okaeiz 

You can use `sendVideo` method for send Gif with animation.

**But this method will soon be committed on the master branch. And I mentioned this issus in commit message.**

Thanks ♥️

**Reply by okaeiz** [reply]
So, in a callback listener function, is it now something like this?
```python
await callback.message.chat.sendVideo(video = "file_id", caption = "This is a GIF")
```

**Reply by kian-ahmadian** [reply]
I think you have not understood the concept of this library in general For example, in the method of sending a message, you only need to have the chat ID and the desired text to send the message.  But in this library, we have tried to remove some parameters with different classifications and it is not necessary to repeat them.  Therefore, *if this method is available in the chat class, you no longer need to know what the event type is!* Therefore, anywhere in this library, if you reach the chat class, you can use its facilities.


### The video sending method is still under development and programming, more commits will be applied soon.

**Reply by okaeiz** [reply]
That was an example though. Thanks for your guide anyway.
